src/auth/is-auth.service.ts

```diff
@@ -55,51 +60,82 @@ export class IsAuthService {
       ]?.replace('Bearer ', '');
 
       const response: Response = httpContext.getResponse();
-      this.appendNewTokensIfRefreshTokenIsValid(
+      return await this.appendNewTokensIfRefreshTokenIsValid(
         refreshToken,
-        accessTokenVerificationResult.user,
+        request,
         response,
+        accessTokenVerificationResult.tokenDecoded,
       );
     }
 
     request.currentUser = this.entityMapper.mapValue(
       CurrentUserWithoutTokens,
-      accessTokenVerificationResult.user,
+      accessTokenVerificationResult.tokenDecoded,
     );
   }
 
-  private async verifyAccessToken(accessToken: string) {
-    if (!accessToken) throw new Unauthorized('Access token not provided');
+  private async verifyAccessToken(accessToken: string): Promise<{
+    isValid: boolean;
+    tokenDecoded?: GeneralTokenDecoded;
+  }> {
+    if (!accessToken) {
+      this.logger.debug('AccessToken cookie not provided');
+      return { isValid: false };
+    }
 
     const accessTokenVerificationResult =
       this.tokenService.checkAccessToken(accessToken);
 
-    const user = await this.userRepo.findOne({
-      where: { id: accessTokenVerificationResult.decodedToken.id },
-    });
-
-    if (!user)
-      throw new Unauthorized("Token's associated id doesn't match any user");
```

Comment:
Querying db for access token verification defies jwt stateless nature with no benefits, as we implemented refresh tokens for a reason

Reply:
completely agree. Thx for updating